framework/src/modules/chain/blocks/chain.js

```diff
@@ -141,16 +141,48 @@ class BlocksChain {
 		if (lastBlock.height === 1) {
 			throw new Error('Cannot delete genesis block');
 		}
-		const previousBlock = await popLastBlock(
-			this.storage,
-			this.interfaceAdapters,
-			this.genesisBlock,
-			this.roundsModule,
-			this.slots,
-			lastBlock,
-			this.exceptions
+		return this.storage.entities.Block.begin('Chain:deleteBlock', async tx => {
+			const previousBlock = await popLastBlock(
+				this.storage,
+				this.interfaceAdapters,
+				this.genesisBlock,
+				this.roundsModule,
+				this.slots,
+				lastBlock,
+				this.exceptions,
+				tx
+			);
+			return previousBlock;
+		});
+	}
+
+	async deleteLastBlockAndStoreInTemp(lastBlock) {
+		if (lastBlock.height === 1) {
+			throw new Error('Cannot delete genesis block');
+		}
+		return this.storage.entities.Block.begin(
+			'Chain:deleteBlockAndStoreInTemp',
+			async tx => {
+				const previousBlock = await popLastBlock(
+					this.storage,
+					this.interfaceAdapters,
+					this.genesisBlock,
+					this.roundsModule,
+					this.slots,
+					lastBlock,
+					this.exceptions,
+					tx
+				);
+				const parsedDeletedBlock = parseBlockToJson(lastBlock);
+				const blockTempEntry = {
+					id: parsedDeletedBlock.id,
+					height: parsedDeletedBlock.height,
+					fullBlock: parsedDeletedBlock,
+				};
+				await this.storage.entities.BlockTemp.create(blockTempEntry, {}, tx);
```

Comment:
Would be nice to have `storage.entities.TempBlock` rather `storage.entities.BlockTemp`